pow-migration/src/lib.rs

```diff
@@ -80,6 +81,48 @@ pub fn get_block_windows(network_id: NetworkId) -> Result<&'static BlockWindows,
     }
 }
 
+pub async fn report_online(
+    pow_client: &Client,
+    block_windows: &BlockWindows,
+    pow_block_number: u32,
+    validator_address: Address,
+) -> Result<(), Error> {
+    // We calculate the range of the online transactions, which start one readiness window (1 day) before the election candidate.
+    let online_start = block_windows.election_candidate - block_windows.readiness_window;
+
+    let online_range = online_start..block_windows.election_candidate;
+
+    // First we obtain the list of online txns that we have sent
+    let txns = get_online_txns(pow_client, validator_address.to_string(), online_range).await;
+
+    let transaction = generate_online_tx(validator_address.to_user_friendly_address());
+
+    // If we have not reported ready yet, we do so:
+    if txns.is_empty() {
+        log::info!("Reporting as online for the first time... ");
+        send_tx(pow_client, transaction).await?;
+    } else {
+        // Get the latest online transaction that was sent
+        let mut latest_online_bn = 0;
+        for txn in txns {
+            if txn.block_number > latest_online_bn {
+                latest_online_bn = txn.block_number;
+            }
+        }
+
+        // We report as online every hour.
+        if pow_block_number > latest_online_bn + 60 {
+            log::info!(
+                "Reporting as online, latest reported block number {}",
+                latest_online_bn
```

Comment:
Use `tracing` style:
```suggestion
            log::info!(
                latest_reported_block_number= latest_online_bn,
                "Reporting as online"
```